### PR TITLE
python310Packages.enaml: add missing input 

### LINKED ISSUE #225896: Build failure: pythonPackages.enaml
### Steps To Reproduce

1. `nix-build '<nixpkgs>' -A pythonPackages.enaml

### Build log

```
error: builder for '/nix/store/30zps21016jqbf9qjmy9wb5gwml8ydg4-python3.10-enaml-0.16.0.drv' failed with exit code 1;
       last 10 log lines:
       > Finished executing setuptoolsBuildPhase
       > installing
       > Executing pipInstallPhase
       > /build/source/dist /build/source
       > Processing ./enaml-0.16.0-cp310-cp310-linux_x86_64.whl
       > Requirement already satisfied: kiwisolver>=1.2.0 in /nix/store/km6wpx1qkdz0mbblr0v52i89r3rhywva-python3.10-kiwisolver-1.4.4/lib/python3.10/site-packages (from enaml==0.16.0) (1.4.4)
       > Requirement already satisfied: bytecode>=0.13.0 in /nix/store/ja3jnd6nan1m6k1rmjb0fdsdlcz26gyl-python3.10-bytecode-0.14.0/lib/python3.10/site-packages (from enaml==0.16.0) (0.14.0)
       > ERROR: Could not find a version that satisfies the requirement pegen>=0.1.0 (from enaml) (from versions: none)
       > ERROR: No matching distribution found for pegen>=0.1.0
       > 
       For full logs, run 'nix log /nix/store/30zps21016jqbf9qjmy9wb5gwml8ydg4-python3.10-enaml-0.16.0.drv'.
```

### Additional context

Fails since 75dcf915a22b01b026fae9a2f /cc @mweinelt

### PR DESCRIPTION
**Pull request by fabaff**
###### Description of changes
Fix build (https://hydra.nixos.org/build/215661711)

Fixes #225896

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 225972` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages failed to build:</summary>
  <ul>
    <li>inkcut</li>
    <li>inkcut.dist</li>
    <li>inkscape-extensions.inkcut</li>
    <li>python310Packages.enamlx</li>
    <li>python310Packages.enamlx.dist</li>
    <li>python311Packages.pegen</li>
    <li>python311Packages.pegen.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.enaml</li>
    <li>python310Packages.enaml.dist</li>
    <li>python310Packages.pegen</li>
    <li>python310Packages.pegen.dist</li>
  </ul>
</details>

**Comment by raboof** [comment]
I'm a bit reluctant to merge this while we can't properly test it, as enamlx and inkcut are in a broken state... I wonder if it'd be better to revert 75dcf915a22b01b026fae9a2fb3ed2fd7de2d96e first, then fix enamlx and inkcut, and only when that works again (or turns out to be necessary) update enaml?

**Review by raboof — APPROVED** [review]
OK, tested it with #226017, seems :+1:

File: pkgs/development/python-modules/enaml/default.nix [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
+, atom
 , buildPythonPackage
+, bytecode
+, cppy
 , fetchFromGitHub
-, setuptools-scm
-, atom
-, ply
 , kiwisolver
+, pegen
```
[the remark this answers is not in the document]
Comment (fabaff) [reply pkgs/development/python-modules/enaml/default.nix:8]: Done

**Review by raboof — APPROVED** [review]

**Pull request merged**